sdk/unity/Assets/Picovoice/Picovoice.cs

```diff
@@ -93,31 +88,52 @@ public class Picovoice : IDisposable
         /// <param name="rhinoSensitivity">
         /// Inference sensitivity. It should be a number within [0, 1]. A higher sensitivity value
         /// results in fewer misses at the cost of(potentially) increasing the erroneous inference rate.
+        /// <param name="requireEndpoint">
+        /// Boolean variable to indicate if Rhino should wait for a chunk of silence before finishing inference.
+        /// </param>
         /// </returns>
-        public static Picovoice Create(string keywordPath, Action wakeWordCallback,
-                                       string contextPath, Action<Inference> inferenceCallback,
-                                       string porcupineModelPath = null, float porcupineSensitivity = 0.5f,
-                                       string rhinoModelPath = null, float rhinoSensitivity = 0.5f)
+        public static Picovoice Create(
+            string accessKey,
+            string keywordPath,
+            Action wakeWordCallback,
+            string contextPath,
+            Action<Inference> inferenceCallback,
+            string porcupineModelPath = null,
+            float porcupineSensitivity = 0.5f,
+            string rhinoModelPath = null,
+            float rhinoSensitivity = 0.5f,
+            bool requireEndpoint = true)
         {
-            Porcupine porcupine = Porcupine.Create(keywordPaths: new List<string> { keywordPath },
-                                                   modelPath: porcupineModelPath,
-                                                   sensitivities: new List<float> { porcupineSensitivity });
+            try
+            {
+                Porcupine porcupine = Porcupine.FromKeywordPaths(accessKey,
+                    keywordPaths: new List<string> { keywordPath },
+                    modelPath: porcupineModelPath,
+                    sensitivities: new List<float> { porcupineSensitivity });
 
-            Rhino rhino = Rhino.Create(contextPath: contextPath,
-                                       modelPath: rhinoModelPath,
-                                       sensitivity: rhinoSensitivity);
+                Rhino rhino = Rhino.Create(accessKey,
+                    contextPath: contextPath,
+                    modelPath: rhinoModelPath,
+                    sensitivity: rhinoSensitivity,
+                    requireEndpoint: requireEndpoint);
+            
+                if (porcupine.FrameLength != rhino.FrameLength)
+                {
+                    throw new PicovoiceInvalidArgumentException("Porcupine and Rhino frame lengths are different");
```

Comment:
always best to print the values e.g. `Porcupine frame length ({0}) and Rhino frame length ({1}) are different`